pkgs/development/python-modules/metaflow/default.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  boto3,
+  requests,
+  setuptools,
+  python,
+}:
+
+buildPythonPackage rec {
+  pname = "metaflow";
+  version = "2.13.9";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Netflix";
+    repo = "metaflow";
+    tag = version;
+    hash = "sha256-f/cWZl59i1PRqNqLV5LtcZoDKdovdCKoQJNxzXKczgs=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    boto3
+    requests
+  ];
+
+  nativeCheckInputs = [
+    requests
+  ];
+
+  preCheck = ''
+    export HOME="$TMPDIR"
+    export USER="metaflow-test-user"
+  '';
+
+  # include all tests but CardExtensionsImportTest
+  # it tests extension installation and require pip
+  checkPhase = ''
+    runHook preCheck
+    cd test/core
+    ${python.interpreter} run_tests.py --num-parallel 8 \
+    --tests FlowOptionsTest,BasicArtifactTest,BasicLogTest,BasicForeachTest,\
+    BasicUnboundedForeachResumeTest,ConstantsTest,BasicParameterTest,ExtensionsTest,\
+    MergeArtifactsPropagationTest,S3FailureTest,BasicUnboundedForeachTest,\
+    MergeArtifactsIncludeTest,ProjectBranchTest,BasicParallelTest,LineageTest,\
+    CardTimeoutTest,TagCatchTest,BasicTagTest,LargeArtifactTest,DetectSegFaultTest,\
+    CardErrorTest,ResumeStartStepTest,CardDecoratorBasicTest,CardWithRefreshTest,\
+    DynamicParameterTest,WideForeachTest,ResumeForeachJoinTest,RunIdFileTest,\
+    CardResumeTest,SecretsDecoratorTest
+    cd ../../
+    CARD_GRAPHS="small-foreach,small-parallel,nested-branches,single-linear-step,simple-foreach"
+    cd test/core && ${python.interpreter} run_tests.py --num-parallel 8 --contexts python3-all-local-cards-realtime --graphs $CARD_GRAPHS && cd ../../
```

What's the error message with this?
```suggestion
  checkPhase = ''
    runHook preCheck

    export HOME="$TMPDIR"
    export USER="metaflow-test-user"

    pushd test/core
    ${python.interpreter} run_tests.py --num-parallel $NIX_BUILD_CORES
    popd

    runHook postCheck
```
Maybe add `--contexts dev-local` if you don't want to run network-dependent tests.

---

`--contexts dev-local` also takes a while to test, I'll use only a small subset of tests then, although all of them passed correctly.

---

Some tests still fail with `--contexts dev-local` but I think fewer are executed overall which is good.

---

We could just `rm tests/card_component_refresh_test.py`.

---

It will still take a while, I left only 2 tests and they take roughly 50s, what time would be acceptable for tests to run?